dspace-swordv2/src/main/java/org/dspace/sword2/VersionManager.java

```diff
@@ -10,75 +10,95 @@
 import org.dspace.authorize.AuthorizeException;
 import org.dspace.content.Bitstream;
 import org.dspace.content.Bundle;
+import org.dspace.content.BundleBitstream;
 import org.dspace.content.Item;
+import org.dspace.content.factory.ContentServiceFactory;
+import org.dspace.content.service.BitstreamService;
+import org.dspace.content.service.BundleService;
+import org.dspace.content.service.ItemService;
 import org.dspace.core.ConfigurationManager;
+import org.dspace.core.Context;
 
 import java.io.IOException;
 import java.sql.SQLException;
 import java.text.SimpleDateFormat;
 import java.util.Date;
+import java.util.Iterator;
+import java.util.List;
 
 public class VersionManager
 {
-	public void removeBundle(Item item, String name)
+	protected ItemService itemService = ContentServiceFactory.getInstance().getItemService();
+	protected BundleService bundleService = ContentServiceFactory.getInstance().getBundleService();
+	protected BitstreamService bitstreamService = ContentServiceFactory.getInstance().getBitstreamService();
+
+	public void removeBundle(Context context, Item item, String name)
 			throws SQLException, AuthorizeException, IOException
 	{
 		boolean keep = ConfigurationManager.getBooleanProperty("swordv2-server", "versions.keep");
-		Bundle[] bundles = item.getBundles(name);
+		List<Bundle> bundles = item.getBundles();
```

Comment:
@KevinVdV , is the list ok here (since the remove call doesn't involve the service)? Or is this another problem spot for concurrent modification exceptions?